### PR TITLE
Fix html version of drawImageNine

### PR DESCRIPTION
**Pull request by flar**
Fixes https://github.com/flutter/flutter/issues/78068

### CODE REVIEW

File: lib/web_ui/lib/src/engine/html/canvas.dart [hunk 1]
```diff
@@ -288,6 +288,47 @@ class SurfaceCanvas implements ui.Canvas {
...
+  // The area from src0 => src1 of the image is painted on the screen from dst0 => dst1
+  // The slices for each dimension are generated independently.
+  List<double> _initSlices(double img0, double imgC0, double imgC1, double img1, double dst0, double dst1) {
+    final double imageDim = img1 - img0;
+    final double destDim = dst1 - dst0;
+
+    if (imageDim == destDim) {
+      // If the src and dest are the same size then we do not need scaling
+      // We return 4 values for a single slice
+      return [ img0, dst0, img1, dst1 ];
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/html/canvas.dart:303]: Let's specify generic arguments. We have a bug in our analysis options to enforce this, but we try to conform to the Flutter style guide for Dart.

**Review by yjbanov — APPROVED** [review]
LGTM if LGT @ferhatb

File: lib/web_ui/lib/src/engine/html/canvas.dart (hunk 1 above)
Comment (flar) [reply lib/web_ui/lib/src/engine/html/canvas.dart:303]: I have that staged to go, but I'm waiting to update the goldens commit hash.

**Review by ferhatb — APPROVED** [review]
Thank you so much cleaner.

**Pull request merged**